system/HTTP/CURLRequest.php

```diff
@@ -542,10 +542,17 @@ protected function setCURLOptions(array $curlOptions = [], array $config = [])
                     throw HTTPException::forInvalidSSLKey($config['ssl_key']);
                 }
 
-                $curlOptions[CURLOPT_CAINFO]         = $file;
-                $curlOptions[CURLOPT_SSL_VERIFYPEER] = 1;
+                $curlOptions[CURLOPT_CAINFO] = $file;
+                if ($config['verify'] === 'yes') {
+                    $curlOptions[CURLOPT_SSL_VERIFYPEER] = 1;
```

Comment:
Why do you set `1`?
It seems this parameter should be boolean.
> CURLOPT_SSL_VERIFYPEER 	
> false to stop cURL from verifying the peer's certificate. Alternate certificates to verify against can be specified with the CURLOPT_CAINFO option or a certificate directory can be specified with the CURLOPT_CAPATH option. 
https://www.php.net/manual/en/function.curl-setopt.php#refsect1-function.curl-setopt-parameters